trilium_alchemy/core/note/note.py

```diff
@@ -965,11 +950,35 @@ def __getitem__(self, key: str | int) -> str | Note:
     def __setitem__(self, key: str | int, value_spec: ValueSpec):
         """
         Create or update attribute with provided name.
+
+        :param key: Attribute name
+        :param value_spec: Attribute value
         """
         self.attributes[key] = value_spec
 
-    def __contains__(self, key: str) -> bool:
-        return key in self.attributes._name_map
+    def __delitem__(self, key: str | int):
+        """
+        Delete attribute with provided name.
+
+        :param key: Attribute name
+        """
+        for attr in self.attributes.owned._name_map[key]:
+            attr.delete()
+        del self.attributes[key]
+
+    def __iter__(self) -> Iterator[list[Attribute]]:
+        """
+        Iterate over attributes.
+
+        :return: Iterator over attributes
+        """
+        yield from self.attributes._name_map
+
+    def __len__(self) -> int:
+        """
+        Number of attributes.
+        """
+        return len(self.attributes)

```

Comment:
In order to be consistent with the other dict semantics of `Note`, I think `len(note)` should give the number of unique owned and inherited attribute names. Basically it should be same as `len(note.keys())`, which is retrieved by `__iter__`, in order to be intuitive IMO. In that case this would be `len(self.attributes._name_map)`, consistent with `__iter__`.